src/VisualStudio/Core/Def/Implementation/Extensions/ServiceProviderExtensions.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.Extensions
+{
+    internal static class ServiceProviderExtensions
+    {
+        public static TInterface GetService<TService, TInterface>(this IServiceProvider serviceProvider)
+        {
+            var service = (TInterface)serviceProvider.GetService(typeof(TService));
+            Debug.Assert(service != null);
```

Comment:
I hadn't really thought about it. I totally just moved this from elsewhere so that I didn't have to copy and paste code.

If `GetService()` returns null, the cast to `TInterface` wouldn't throw unless it's a value type, so the assert is legit.

For your amusement:

```C#
        static T Foo<T>(object o)
        {
            var x = (T)o;
            return x;
        }

        static void Main(string[] args)
        {
            object o = null;
            var x = Foo<string>(o); // does not throw
            var y = Foo<int>(o); // throws a NullReferenceException
        }
```